src/GovUk.Education.ExploreEducationStatistics.Admin/Services/DataGuidanceService.cs

```diff
@@ -1,150 +1,157 @@
 using GovUk.Education.ExploreEducationStatistics.Data.Model.Database;
 using GovUk.Education.ExploreEducationStatistics.Data.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Data.Services.ViewModels;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.EntityFrameworkCore;
-using static GovUk.Education.ExploreEducationStatistics.Admin.Validators.ValidationUtils;
+using static GovUk.Education.ExploreEducationStatistics.Admin.Validators.ValidationErrorMessages;
+using static GovUk.Education.ExploreEducationStatistics.Common.Validators.ValidationUtils;
 
 namespace GovUk.Education.ExploreEducationStatistics.Admin.Services
 {
     public class DataGuidanceService : IDataGuidanceService
     {
         private readonly ContentDbContext _contentDbContext;
-        private readonly IPersistenceHelper<ContentDbContext> _contentPersistenceHelper;
-        private readonly IDataGuidanceSubjectService _dataGuidanceSubjectService;
+        private readonly IDataGuidanceDataSetService _dataGuidanceDataSetService;
         private readonly StatisticsDbContext _statisticsDbContext;
         private readonly IUserService _userService;
-        private readonly IReleaseDataFileRepository _fileRepository;
+        private readonly IReleaseDataFileRepository _releaseDataFileRepository;
 
         public DataGuidanceService(ContentDbContext contentDbContext,
-            IPersistenceHelper<ContentDbContext> contentPersistenceHelper,
-            IDataGuidanceSubjectService dataGuidanceSubjectService,
+            IDataGuidanceDataSetService dataGuidanceDataSetService,
             StatisticsDbContext statisticsDbContext,
             IUserService userService,
-            IReleaseDataFileRepository fileRepository)
+            IReleaseDataFileRepository releaseDataFileRepository)
         {
             _contentDbContext = contentDbContext;
-            _contentPersistenceHelper = contentPersistenceHelper;
-            _dataGuidanceSubjectService = dataGuidanceSubjectService;
+            _dataGuidanceDataSetService = dataGuidanceDataSetService;
             _statisticsDbContext = statisticsDbContext;
             _userService = userService;
-            _fileRepository = fileRepository;
+            _releaseDataFileRepository = releaseDataFileRepository;
         }
 
-        public async Task<Either<ActionResult, DataGuidanceViewModel>> Get(Guid releaseId)
+        public async Task<Either<ActionResult, DataGuidanceViewModel>> GetDataGuidance(Guid releaseId,
+            CancellationToken cancellationToken = default)
         {
-            return await _contentPersistenceHelper.CheckEntityExists<Release>(releaseId)
-                .OnSuccessDo(release => _userService.CheckCanViewRelease(release))
-                .OnSuccess(BuildViewModel);
+            return await _contentDbContext.Releases
+                .FirstOrNotFoundAsync(release => release.Id == releaseId, cancellationToken)
```

Comment:
Just interested in why you're doing this instead of using `CheckEntityExists`? I assume because of the cancellation token?

Reply:
It's because we've recently been talking about phasing out our use of `PersistenceHelper` due to it not adding extra value over what can already be achieved using Entity Framework ootb methods like `FirstOrDefaultAsync` combined with our own extensions like `FirstOrNotFoundAsync`. This combines `FirstOrDefaultAsync` and our own `NotFound`. 

Using `PersistenceHelper` also requires wiring an extra dependency.

We have a note about it already raised for the back end style guide meeting.